pkg/image/skopeo.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2018 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"kubevirt.io/containerized-data-importer/pkg/system"
+)
+
+const dataTmpDir string = "/data_tmp"
+const whFilePrefix string = ".wh."
+
+// SkopeoOperations defines the interface for executing skopeo subprocesses
+type SkopeoOperations interface {
+	CopyImage(string, string, string, string) error
+}
+
+type skopeoOperations struct{}
+
+type manifest struct {
+	Layers []layer `json:"layers"`
+}
+type layer struct {
+	Digest string `json:"digest"`
+}
+
+var skopeoExecFunction = system.ExecWithLimits
+
+var processLimits = &system.ProcessLimitValues{AddressSpaceLimit: maxMemory, CPUTimeLimit: maxCPUSecs}
+
+var skopeoInterface = NewSkopeoOperations()
+
+// NewSkopeoOperations returns the default implementation of QEMUOperations
```

Comment:
in the comment - instead 'QEMUOperations', 'SkopeoOperations'?

Reply:
done.